### PR TITLE
CLN: F-string formatting in pandas/_libs/*.pyx

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by MomIsBestFriend** [comment]
Super lost,
the tests that failed before are now passing, and the test that passed before are now failing, all I changed was 5 spaces before a string to make pep8 happy.

I have no clue what am I doing wrong.

can anyone help?

File: pandas/_libs/interval.pyx [hunk 1]
```diff
@@ -381,17 +379,15 @@ cdef class Interval(IntervalMixin):
 
         left, right = self._repr_base()
         name = type(self).__name__
-        repr_str = '{name}({left!r}, {right!r}, closed={closed!r})'.format(
-            name=name, left=left, right=right, closed=self.closed)
+        repr_str = f'{name}({left}, {right}, closed={self.closed})'
```
Comment (jbrockmendel) [inline pandas/_libs/interval.pyx]: i think the problem is here, where you are dropping the "!r", which loses single-quotes around the affected values

Comment (MomIsBestFriend) [reply pandas/_libs/interval.pyx]: Oh, so it should be:
```f'{name}{left!r}, {right!r}......'  ```

**Comment by MomIsBestFriend** [comment]
@jbrockmendel You legend<3

**Comment by jreback** [comment]
thanks @MomIsBestFriend 

if you can make a master issue for this; would like to do the entire codebase but in multiple PRs, then add a check that we don't regress these changes.

**Pull request merged**